File: app/src/main/java/com/example/pam_app/repository/IncomeMapper.java
package com.example.pam_app.repository;

import com.example.pam_app.db.IncomeEntity;
import com.example.pam_app.model.Income;
import com.example.pam_app.model.IncomeType;

public class IncomeMapper {

    public Income toModel(final IncomeEntity incomeEntity) {
        return new Income(incomeEntity.getId(), incomeEntity.getTitle(), incomeEntity.getAmount(),
                IncomeType.values()[incomeEntity.getIncomeType()], incomeEntity.getDate());
    }

    public IncomeEntity toEntity(final Income income) {
        return new IncomeEntity(income.getTitle(), income.getAmount(),
                income.getIncomeType().ordinal(), income.getDate());
    }
}

File: app/src/main/java/com/example/pam_app/repository/IncomeMapper.kt
package com.example.pam_app.repository

import com.example.pam_app.db.IncomeEntity
import com.example.pam_app.model.Income
import com.example.pam_app.model.IncomeType
import kotlin.jvm.Synchronized

class IncomeMapper {
    fun toModel(incomeEntity: IncomeEntity?): Income {
        return Income(incomeEntity?.id!!, incomeEntity.title, incomeEntity.amount,
                IncomeType.values()[incomeEntity.incomeType], incomeEntity.date)
    }

    fun toEntity(income: Income): IncomeEntity {
        return IncomeEntity(income.comment, income.amount,
                income.incomeType.ordinal, income.date)
    }
}